### DISCUSSION TITLE
-mpi won't compile

### DISCUSSION THREAD

**Discussion by bergeson00** [PrincetonUniversity/athena#508] [Q&A]
I'm a newbie, working through the Athena++ tutorial. Maybe this is a linux problem? I'm running into a problem on the "Running 3D MHD with OpenMP and MPI" page, https://github.com/PrincetonUniversity/athena/wiki/Running-3D-MHD-with-OpenMP-and-MPI. When I compile the code, 
```
> cd ~/athena
> python configure.py --prob blast -b --flux hlld -mpi
> make clean
> make
```
I get the errors
```
undefined reference to `secure_getenv@GLIBC_2.17'
undefined reference to `clock_gettime@GLIBC_2.17'
ld returned 1 exit status
make: *** [Makefile:96: bin/athena] Error 1
```
Any guidance for resolving this? I'm using Ubuntu 22.04.2 LTS, and it is a fairly new installation.

**Comment by felker** [comment]
Does it compile individual `*.o` files in the `obj/` directory but then fails at the final linking step?

Are you able to compile and run a "hello world" C code on your platform successfully?

**Reply by bergeson00** [reply]
Apparently  not. I tried this code:
```
#include <mpi.h>
#include <stdio.h>

int main ( int argc, char *argv[]) {

  int rank, size;

  MPI_Init (&argc, &argv); 

  MPI_Comm_size(MPI_COMM_WORLD, &size); 
  MPI_Comm_rank(MPI_COMM_WORLD, &rank); 

  printf ("Hellow world from rank %d\n", rank);
  if (rank == 0) printf("MPI World size = %d processes\n", size);

  MPI_Finalize(); 

  return 0;
}
```
tried to compile with the command
```
mpicc testingmpi.cpp
```
and received the same errors,
```
libgfortran.so.4: undefined reference to 'clock_gettime@GLIBC_2.17'
libgfortran.so.4: undefined reference to 'secure_getenv@GLIBC_2.17'
```
Apparently this is not an athena problem, but something in my environment.
